src/main/java/com/google/error_analyzer/backend/LogDao.java

```diff
@@ -102,15 +102,21 @@
     }
 
     //search db using regex and keywords and store back in db searchHits
-    //  sorted by logLineNumber
+    //Returns name of the new error File
     @Override 
-    public boolean errorQuery(String fileName) throws IOException {
-        return true;
-    }
-
-    //store sorted identified errors back in database
-    public void storeErrorLogs(String fileName, SearchHits hits) throws IOException {
-        return;
+    public String errorQuery(String fileName) throws IOException {
+        BooleanQuery booleanQuery = new BooleanQuery();
+        SearchRequest searchRequest = booleanQuery.createSearchRequest(fileName);
+        SearchResponse searchResponse = client
+            .search(searchRequest, RequestOptions.DEFAULT);
+        SearchHits hits = searchResponse.getHits();
+        String errorFile = fileName.concat("error");
+        for (SearchHit hit : hits) {
+            String jsonSource =  hit.getSourceAsString();
+            String id = hit.getId();
+            storeLogLine(errorFile, jsonSource, id);
+        }
+        return errorFile;
     }
```

Comment:
errorQuery function should be in DaoHelper since it contains logic more than required for fetching from elasticsearch.
Dao should only have function like fetching results from elasticsearch.

Reply:
I have a slight preference to put it in LogDao.